### DISCUSSION TITLE
console.log() in the browser

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#17475] [Questions]
I have this code:

```html
<script setup lang="ts">
console.log('print me')
</script>
```

but I see nothing in my browser console. Is there a setting somewhere that I need to enable?

I'm running: v3.0.0-rc.3

**Comment by FSL89** [comment]
Where have you put the script tag? If it's in a file under **pages**, you could try restarting the app. I sometimes have to reload with clearing cache using **CTRL** + **SHIFT** + **R** before some changes are applied.

**Reply by ghost** [reply]
It is in a file under pages.
I have tried restarting, and I have tried to clear cache. None of that seem to have any effect.
I can see the log output in the terminal of my developer machine.... but not in the browser.

**Reply by FSL89** [reply]
Are you perhaps filtering log types in dev tools?

**Reply by ghost** [reply]
No filtering in the browser console...

**Comment by TheAlexLichter** [comment]
Hey 👋 
Can't reproduce this in e.g. StackBlitz. See [my example](https://stackblitz.com/edit/nuxt3-discussions-5220?file=app.vue).
![grafik](https://user-images.githubusercontent.com/640208/171195116-0f65218d-a64b-4534-8206-b819824710c8.png)

**Comment by sewalsh** [comment]
This won't show in your browser console if you have SSR enabled and it's the first request. It will appear in the server console/terminal output. However if you navigate client-side and return to the page, it will show any console.log in the browser console.

**Reply by ghost** [reply]
I have tried this theory but it made no difference. I tried to do the same in a clean vue3 project. Here it works as expected.

**Reply by sewalsh** [reply]
vue3 doesn't do SSR. Nuxt does. That's the key difference.

**Reply by ghost** [reply]
So then the question becomes how best to debug if console.log is not available in browser console?

**Reply by danielroe** [reply]
`console.log` should work in your browser. Please see if you can create a minimal reproduction on the link @manniL shared.

**Comment by howbizarre** [comment]
> I have this code:
> 
> ```
> <script setup lang="ts">
> console.log('print me')
> </script>
> ```
> 
> but I see nothing in my browser console. Is there a setting somewhere that I need to enable?
> 
> I'm running: v3.0.0-rc.3

If you write this code in file on the server/api or server/middleware directory, you can see the output in the Node console.
Where you start: yarn/npm run dev
